pyiron_base/job/jobtype.py

```diff
@@ -95,8 +98,50 @@ def __new__(
             job.set_input_to_read_only()
         return job
 
+    @classmethod
+    def un_register_job_type(cls, job_name):
+        if job_name in cls._job_class_dict:
+            del cls._job_class_dict[job_name]
+        else:
+            raise KeyError(f"No JobType with name '{job_name}' found.")
+
+    @classmethod
+    def register_job_type(cls, cls_name, _cls):
+        if cls_name is None:
+            return
+        elif cls_name not in cls._job_class_dict and isinstance(_cls, str):
+            cls._job_class_dict[cls_name] = _cls
+        elif (
+            cls_name in cls._job_class_dict
+            and isinstance(_cls, str)
+            and cls._job_class_dict[cls_name] == _cls
+        ):
+            pass
+        elif cls_name not in cls._job_class_dict and not isinstance(_cls, str):
+            if cls_name != _cls.__name__:
+                raise NotImplementedError(
+                    "Currently, the given name has to match the class name."
+                )
+            cls._job_class_dict[cls_name] = cls._get_class_path_str(_cls)
+        elif (
+            cls_name in cls._job_class_dict
+            and not isinstance(_cls, str)
+            and cls._job_class_dict[cls_name] == cls._get_class_path_str(_cls)
+        ):
+            pass
+        else:
+            raise ValueError(
+                f"A JobType with name {cls_name} is already defined! New class = {repr(_cls)}, "
+                f"already registered class = {repr(cls._job_class_dict[cls_name])}."
+            )
+
+    @staticmethod
+    def _get_class_path_str(_cls):
+        full_path = repr(_cls).split("<class '")[-1].split("'>")[0]
+        return ".".join(full_path.split(".")[:-1])
```

Comment:
Use `_cls.__name__` and `_cls.__module__` instead of fiddling with strings.

Reply:
Yes, that full function only returns the value of `cls.__module__` if I am not mistaken 😲 I was not aware of that attribute and copied the function from some place in the `hdf` modules... Will remove the entire function 😄